### PR TITLE
Nuvoton NCT6798D Improvements, ROG Strix Z790-E Gaming Wi-Fi added

### PR DESCRIPTION
**Pull request by Xcalibur37**
Made multiple updates: 

1. Nct677XX.cs - Nuvoton NCT6978D - Rescaled voltages to properly display values over 2.048v.
2. SuperIOHardware.cs: Added specification for ROG_STRIX_Z790_E_GAMING_WIFI.

### CODE REVIEW

**Comment by Xcalibur37** [comment]
Apologies for the weird check-ins. I was trying to force git to do a proper line-by-line compare on Nct677X.cs but it didn't seem to understand and makes it harder to eyeball the changes via diff. Biggest change here is Lines 437 - 496. I added specific voltage scaling for the NCT6796D family. I would recommend testing against another in the family to ensure things look kosher. I took this explicitly from the NCT6798D datasheet.

**Comment by Tellinq** [comment]
I wanted to come in here - mainly to ask the same questions as https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/1450#issuecomment-2337020081

primarily Index 10 - Memory Controller. I’ve been testing with a friend on the Z790-A and intend to add it later on.

Mind also checking the temperature sensors past the auxiliary sensors as well? Just to make sure the Chipset and a proper CPU temperature is accounted for.

As for you mentioning NCT6796D - I have an ASRock Z690 Extreme that I can test against this PR and see if there’s any notable changes and corrections I’ll need to make for the data entries I’ve done for the specific boards I added

**Comment by Tellinq** [comment]
As for the Z790-E entry: It would be much preferred if the Fan controls were properly named instead of the generic Fan 1-7 Namings. Preferably CPU Fan, CPU Optional Fan, Chassis Fan `#`, AIO Pump, etc.

**Note I’m not a maintainer, but I do tend to care when people make data entries and I’d prefer to see more complete entries**

**Comment by Xcalibur37** [comment]
> I wanted to come in here - mainly to ask the same questions as [#1450 (comment)](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/1450#issuecomment-2337020081)
> 
> primarily Index 10 - Memory Controller. I’ve been testing with a friend on the Z790-A and intend to add it later on.
> 
> Mind also checking the temperature sensors past the auxiliary sensors as well? Just to make sure the Chipset and a proper CPU temperature is accounted for.
> 
> As for you mentioning NCT6796D - I have an ASRock Z690 Extreme that I can test against this PR and see if there’s any notable changes and corrections I’ll need to make for the data entries I’ve done for the specific boards I added

Good stuff. Any QC we can do would be great. I spent a lot of time on the NCT6796D data sheet to understand the values so I am hoping the adjustments I made are accurate and provide more reasonable expectations for these readings (e.g. the 12V reading you see in HWInfo).

**Comment by Xcalibur37** [comment]
Here is a visual on how the readings look for the Motherboard:

Untitled

**Comment by Xcalibur37** [comment]
> As for the Z790-E entry: It would be much preferred if the Fan controls were properly named instead of the generic Fan 1-7 Namings. Preferably CPU Fan, CPU Optional Fan, Chassis Fan `#`, AIO Pump, etc.
> 
> **Note I’m not a maintainer, but I do tend to care when people make data entries and I’d prefer to see more complete entries**

Yeah, for sure - I was not able to accurately derive this data as I use Corsair Commander Core XT's to control my fans so it's all 0. Makes it hard to correlate which fan represents which fan header.

**Comment by Tellinq** [comment]
> Here is a visual on how the readings look for the Motherboard:
> 
> Untitled

Appreciate it. I’ll check in with my friend and use this PR to see if the Memory Controller properly aligns. 

Either way thank you so much for your work on this PR!

**Comment by Tellinq** [comment]
Oh! Also wanted to bring up - in that PR I referenced - the VRM temperature was added to the EC, but never was added to the Z790-E due to no ability to test that board. Mind adding that in this PR since there’s some sensors in the EC for your board already added aside from VRM?

Just also wanting to make sure that that VRM temperature is also showing not just for that one board that was implemented in the prior PR.

**Comment by Xcalibur37** [comment]
> Oh! Also wanted to bring up - in that PR I referenced - the VRM temperature and CPU Optional Fan was added to the EC, but never was added to the Z790-E due to no ability to test that board. Mind adding that in this PR since there’s some sensors in the EC for your board already added aside from VRM?
> 
> Just also wanting to make sure that that VRM temperature is also showing not just for that one board that was implemented in the prior PR.

I will take a look at this. In the meantime, I am going to test the fan headers to see if I can derive the real values. The annoying part is that there are technically 8 headers (as per Armoury Crate) but 6798D only exposes 7. I am expecting it will be similar to the Dark Hero.

**Comment by Xcalibur37** [comment]
Fan headers are now properly named. The one that is missing is CPU OPT Fan. This one is not exposed (as far as I can tell) by the 6798D.

image

**Comment by Tellinq** [comment]
From my understanding the CPU Optional header is seen on the Embedded Controller. I believe if you just clone what was done in the PR I previously referenced you should be able to see the changes

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (Tellinq) [inline LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:3850]: Mind checking Temperature Index 13 and 22? Index 13 should be the Chipset, while 22 is another (and typically more responsive) COU temperature

A word on Auxiliary temperature sensors: Those tend to not actually include any meaningful values so I recommend removing those sensors. An obvious giveaway is a temperature never changing, or values going below ambient. Based off all your screenshots all should probably be removed.

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (Tellinq) [inline LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:3824]: You might get notified that this should get renamed to `Vcore` instead of this. Although what you put makes a bit more sense to name this way in my opinion.

We'll see what PhyxionNL thinks

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (Tellinq) [inline LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:3829]: Based off the ASUS boards I've tested including the Z790-A, this tends to be `IVR Atom L2 Cluster #1`. Worth mentioning since this entire data entry is super similar to the Z790-A and Dark Hero layout.

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (Tellinq) [inline LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:3833]: VTT -> `CPU Termination`

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (Tellinq) [inline LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:3838]: Nit pick here - `CPU Input Auxiliary` to be a tiny bit more specific

**Review by Tellinq — CHANGES_REQUESTED** [review]

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (Xcalibur37) [reply LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:3838]: No problem. I will change that.

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (Xcalibur37) [reply LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:3833]: I did see that correction on other entries post release. I will correct this.

**Comment by Xcalibur37** [comment]
> From my understanding the CPU Optional header is seen on the Embedded Controller. I believe if you just clone what was done in the PR I previously referenced you should be able to see the changes

You are correct. Making the changes you notes from the PR, I can now see this value.

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (Xcalibur37) [reply LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:3850]: Agreed. Let me see what I can dig up.

**Comment by Xcalibur37** [comment]
> > From my understanding the CPU Optional header is seen on the Embedded Controller. I believe if you just clone what was done in the PR I previously referenced you should be able to see the changes
> 
> You are correct. Making the changes you noted from the PR, I can now see this value.

image

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (Xcalibur37) [reply LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs]: Will do.

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (Xcalibur37) [reply LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs]: image

**Comment by Xcalibur37** [comment]
By the way, I share this to amuse you - I literally opened my case and plugged a fan into each header to verify it aligned to the correct fan address and label in Armoury Crate. The things we do to get it right :).

**Comment by Xcalibur37** [comment]
Might sound like a bit of work but among the many cleanups we could do - should we consider writing a set of consts for these labels and replacing the string references so the labeling is more consistent across boards?

**Comment by Tellinq** [comment]
> Might sound like a bit of work but among the many cleanups we could do - should we consider writing a set of consts for these labels and replacing the string references so the labeling is more consistent across boards?

Either that, or we may need to consider Locale in a future PR. I know some other programs like HWiNFO64 account for this. I feel that achieving this would likely be a more ideal solution for consistency and translation purposes.

**Comment by Tellinq** [comment]
> By the way, I share this to amuse you - I literally opened my case and plugged a fan into each header to verify it aligned to the correct fan address and label in Armoury Crate. The things we do to get it right :).

Oh yeah! Been there with my personal boards. Definitely a hassle when physically changing headers but hey if it’s what’s needed, so be it

File: LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/Nct677X.cs
Comment (Tellinq) [inline LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/Nct677X.cs:1]: Not sure how this entire reformat happened. Had to use a diff checker site to find the real differences. Not sure what PhyxionNL will want from this but might be worth rolling back to a state prior to this PR, and then recommit to reflect your actual changes.

Comment (Xcalibur37) [reply LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/Nct677X.cs:1]: Yeah, I mentioned it earlier. Git got really stupid with me on that file. If we can't salvage it, let me know and I will grenade it and redo the work. It wasn't that much really and I know the changes fairly well.

**Comment by Tellinq** [comment]
Before I get to testing I noticed some tapers worth mentioning:

The rescale you did is technically only applicable for Z690 and Z790 on ASUS boards. Especially on the IMC VDD. The problem here is that on DDR4 boards, this is actually just DIMM voltage which is doubled instead of 2.22

On NCT6796D family, the Vcore is typically fine as is. As of right now in accordance to the current global data entry on ASRock boards, the VCore is doubled until someone like myself makes separate data entries correcting that. I'll show how my Z690 Extreme appears via this PR once I have time for testing

I also have an X570-Plus WiFi I can boot up to test with at some point. I believe this also might stem with the VCore scaling you did correctly.

May be worth visiting my PRs for data entries I did for perspective: #1451 and #1362 

Again, this is based off a glimpse I noticed in the diff checker. I’ll make sure I validate my concerns later on. If this is true, current data entries will need to be modified or at the very least narrow down the scope of the change to reflect specific kinds of boards (such as Z600/Z790)

**Comment by Tellinq** [comment]
On a positive side note: thankfully this PR should address #1432

**Comment by Xcalibur37** [comment]
> Before I get to testing I noticed some tapers worth mentioning:
> 
> The rescale you did is technically only applicable for Z690 and Z790 on ASUS boards. Especially on the IMC VDD. The problem here is that on DDR4 boards, this is actually just DIMM voltage which is doubled instead of 2.22
> 
> On NCT6796D family, the Vcore is typically fine as is. As of right now in accordance to the current global data entry on ASRock boards, the VCore is doubled until someone like myself makes separate data entries correcting that. I'll show how my Z690 Extreme appears via this PR once I have time for testing
> 
> I also have an X570-Plus WiFi I can boot up to test with at some point. I believe this also might stem with the VCore scaling you did correctly.
> 
> May be worth visiting my PRs for data entries I did for perspective: #1451 and #1362
> 
> Again, this is based off a glimpse I noticed in the diff checker. I’ll make sure I validate my concerns later on. If this is true, current data entries will need to be modified or at the very least narrow down the scope of the change to reflect specific kinds of boards (such as Z600/Z790)

I was afraid of that. I am interested to see what you find. I didn't have a broad enough data set to verify some of my claims. Some of the doubling is directly referenced in the Nuvoton Data Sheet for 6798D (with a very dumb equation that literally says "we took your value and cut it in half" if you reduce it).

Curious to see what you find - happy to refine any way we see fit.

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs
Comment (PhyxionNL) [reply LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:3824]: CPU Core would be OK, but then I prefer to see it renamed everywhere. I don't see a point in naming things double (a voltage is already a voltage :))

**Comment by Tellinq** [comment]
Here's what I think we're going to need to do next:

In the SuperIO class:

These chips:
- NCT6796D 
- NCT6796DR 
- NCT6797D 
- NCT6798D 
- NCT6799D

Will need to be separate to the other SuperIO chips in the current cluster. The main challenge here is identifying which board corresponds to which chip as some weren't properly commented on.

> Some of the doubling is directly referenced in the Nuvoton Data Sheet for 6798D
I for whatever reason cannot find this data sheet online, but I can find 6796D. Unless there was a mixup here, mind sending this data sheet so I can take a look for reference?

Also, back on the Vcore, I noticed NCT6799D on ASUS boards also scales with 1.11 (See #1447), but with ASRock it's the same story as NCT6796D where the value is fine as-is (See #1420) (unfortunately by default it's doubled, but once those chips I mentioned are split this can easily be addressed globally). My guess is that this is universal across NCT6796D-NCT6799D on ASUS and ASRock. 

I can go ahead and check each board once I get time to test my boards in particular to verify my claims. I can PR some changes to your fork, although GitHub will be a bit picky on that since I technically can only use one fork per account. If you don't mind adding me to be able to write changes to your fork (I'll PR, not directly push to your master branch)

**Comment by Xcalibur37** [comment]
I apologize for assuming a generalization on this group of chips. I guess it's more nuanced as you pointed out. You are right on the 6798 data sheet - it doesn't exist. All documentation and references I have seen point back to 6796 as the base data sheet for all of these boards. I had a hard time believing this for awhile because I would have thought 6798 would be able to actually provide DIMM temps from the socket like it appears we see in HWInfo and HWMonitor - but no matter what I did in my experiments it yielded a reading of "0". Also, the 5V and 12V readings were suspect until I finally realized they were scaling down to cap values at 2.048V and had to do VIN# comparisons between tools to derive what we were actually looking at. It was a real chore to say the least.

If you do have an opportunity to compare and contrast against a few other variants it would be greatly appreciated. I would if I had those chips in my lab but I am fairly limited with my hardware (I currently have that Z790-E ASUS, and a Z170-A on hand though I could get access to a Z490 ASUS if needed).

Another thing not related to this case but worth mentioning - I think we need to get access to something running on a Core Ultra - I suspect it's not reading as a proper Meteor Lake and limiting processor readings for users. It's in a separate Issue that I was working to assist on but I don't have access to the hardware.

**Comment by Tellinq** [comment]
image
Just tested my Z690 Extreme - Vcore as I suspected isn't correct anymore (righht is before, left is your build for my board)

If this is going to work, we'll need to have the current data entries modified to reflect the scaling changes. - Vcore however should only have the 1.11 multiplier on ASUS boards as that's where it's applicable

I'll get my X570-Plus WiFi for confirmation, just will take a bit since it's locked in one of my closets

**Comment by Xcalibur37** [comment]
> image Just tested my Z690 Extreme - Vcore as I suspected isn't correct anymore (righht is before, left is your build for my board)
> 
> If this is going to work, we'll need to have the current data entries modified to reflect the scaling changes. - Vcore however should only have the 1.11 multiplier on ASUS boards as that's where it's applicable
> 
> I'll get my X570-Plus WiFi for confirmation, just will take a bit since it's locked in one of my closets

Nice - I can add that to the scaling criteria in Nct67XX.cs. We would still need to determine which specific Nuvoton's it impacts. Ex. does my change really only apply to NCT6798D or does it in fact impact all of those? Is it potentially restricted to 600 and 700 series based on your observations?

It's just surprising to me how much manufacturers play with these numbers. Makes consistent readings very difficult to get right.

**Comment by Tellinq** [comment]
Here’s what I recommend:

1. Make the criteria where only ASUS boards with these chips have the 1.11 multiplier
2. Limit IMC VDD to Intel 400/500/600/700 series keep in mind DDR4 boards simple need a double, instead of 2.22
3. Limit CPU L2 to Intel 600/700
4. Only double voltage index 14 on ASUS boards. 9 and 13 can stay as-is
5. Limit the 5V scaling changes on NCT9697D, NCT9698D, and NCT9699D for ASUS and MSI. ASRock doesn’t apply the same scaling (based off the current additions on all manual data entries so far). All the 3.3V and 12V changes do apply to the entire family.

**Comment by Xcalibur37** [comment]
If your 12V is running at 144V - get out of the house now!

**Comment by Tellinq** [comment]
Oh for sure!

I’ve made changes on my end to change the current data entries to reflect these scaling changes. I’ll PR the changes probably on a separate account so I can actually fork your fork

**Comment by Xcalibur37** [comment]
> Oh for sure!
> 
> I’ve made changes on my end to change the current data entries to reflect these scaling changes. I’ll PR the changes probably on a separate account so I can actually fork your fork

So, I starred your changes via email. Given what we have been discussing - do you think I should blast this fork into oblivion and redo it so we get the right diff compares on Nct67xx.cs? Then we can make the proper labeling updates you both suggested and bring in these scalers to be more contextually relevant to the affected manuf, chip, series. Or, should we move on with this and get it where we want it?

Your call - I am happy to roll with whatever we decide.

**Comment by Tellinq** [comment]
Let's continue moving on with this PR. No need to blast it into oblivion. I just made my PR to split the NCT list in SuperIO. We will now need to just make sure the NCT class reflects the proposed scaling changes.

**Comment by Xcalibur37** [comment]
Updated as per your merge request. Let me know what I can change other than the labeling suggested for VCore.

**Review by Tellinq — COMMENTED** [review]
I'm going to act a bit stingy here: While these changes definitely will be useful... you sure you're going to want them within this PR? My suggestion just for the sake in the case this PR needs more changes that you move your changes in this new commit to a separate branch, and then PR for those new changes accordingly (especially since none of the files changed in this commit were changed prior during the lifecycle of this PR).

Usually a good idea to keep changes within scope instead of piling a load of changes that have multiple goals.

**Comment by Xcalibur37** [comment]
All good. I will take care of that now.

**Comment by Xcalibur37** [comment]
This is back to where it was. Opening a new pull request to a branch of my repo.

**Comment by PhyxionNL** [comment]
> Usually a good idea to keep changes within scope instead of piling a load of changes that have multiple goals.

+1, please separate this PR into multiple, this is very large to review like this.

**Pull request closed without merging**
